package.json

```diff
@@ -68,16 +100,31 @@
     "grunt-plato": "0.1.4",
     "grunt-preprocess": "4.0.0",
     "grunt-template": "0.2.3",
```

Comment:
After talking with @jasonLaster, we decided this can be handled in 3 parts: 1. adding gulp, 2. converting src to es6 modules then 3. removing grunt. I agreed with doing it that way cause if things broke you could still run it with grunt. Also, I think the initial adding gulp PR was supposed to be minimal but it was easy to add most of the tasks so I did (though **not all** of the old tasks have been converted like coverage which is currently broken in grunt as well). 

If people think we can remove grunt here as well, I can do that too.